src/wallet/wallet.cpp

```diff
@@ -70,6 +71,47 @@ const CWalletTx* CWallet::GetWalletTx(const uint256& hash) const
     return &(it->second);
 }
 
+// Generate a new spending key and return its public payment address
+CZCPaymentAddress CWallet::GenerateNewZKey()
+{
+    AssertLockHeld(cs_wallet); // mapZKeyMetadata
+    auto k = SpendingKey::random();
+    auto addr = k.address();
+
+    // Check for collision, even though it is unlikely to ever occur
+    if (CCryptoKeyStore::HaveSpendingKey(addr))
+        throw std::runtime_error("CWallet::GenerateNewZKey(): Collision detected");
```

Comment:
I don't understand this check. The only way I can think of for an address collision is if:

a. We use a deterministic PRNG for key generation and there was some state rollback,
b. Our key scheme is vulnerable to collisions,
c. Our address scheme is vulnerable to collisions.

Did I miss any? Either b or c would imply a cryptographic vulnerability, in which case we should fix that vulnerability rather than add code to respond to the vulnerability.

Case a. is an important possibility (especially given many virtualized environments that may take a snapshot, run after the snapshot, then resume from a previous snapshot). This only matters if we're doing deterministic key generation. Are we?

Another thought experiment: How do you exercise that code in a unit test to achieve thorough branch coverage?